lib/api/_loaders/element-global.js

```diff
@@ -89,7 +90,23 @@ class ElementGlobal {
 
     const byLocator = Locator.create(locator);
 
-    this.element = await this.transport.driver.wait(until.elementLocated(byLocator), this.timeout, null, this.retryInterval);
+    this.setPropertiesFromElement(locator);
+
+    const elements = await this.transport.driver.wait(until.elementsLocated(byLocator), this.timeout, null, this.retryInterval);
+
+    if (elements.length === 0) {
+      throw new Error(`Element ${byLocator} was not found.`);
+    }
+
+    if (typeof(this.index) != 'number'){
+      throw new Error(`Index ${this.index} is not of type number for locator: ${byLocator}`);
+    }
+
+    if (elements.length < this.index){
```

Comment:
```suggestion
    if (elements.length <= this.index){
```

Since the index starts from 0.